### PR TITLE
Weekly openmeteo load.py: Alex

### LINKED ISSUE #15: Create test file for weekly astronomy load script
As a developer, I want to test the load.py script to ensure that the cleaned and transformed weather data is correctly inserted into the database, so I can trust that the pipeline loads data accurately for future retrieval by users.

### PR DESCRIPTION
**Pull request by AlexOBarnes**
This closes #15 #21 #27 

What I have done:
- Created `test_load.py`
- Written tests for `load.py`
- Created `load.py` and implemented passing functions
- Dockerised entire pipeline

### CODE REVIEW

File: weekly-openmeteo/Dockerfile
Comment (olliejth) [inline weekly-openmeteo/Dockerfile:1]: We should standardise all the lambda handler functions to be named in this way for convenience going forward.

File: weekly-openmeteo/conftest.py
Comment (olliejth) [inline weekly-openmeteo/conftest.py:1]: Nice use of fixtures.

File: weekly-openmeteo/load.py
Comment (olliejth) [inline weekly-openmeteo/load.py:1]: Could the empty nested list be instantiated inside the function here?

This way the function could just be called using: "load_data()"

Comment (olliejth) [inline weekly-openmeteo/load.py:1]: Is the empty nested list here just a placeholder?

**Review by olliejth — APPROVED** [review]
Solid code, LGTM.

File: weekly-openmeteo/test_load.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import pytest
+from unittest import mock
+from load import load_data
+
+
```
Comment (MurasatoNaoya) [inline weekly-openmeteo/test_load.py:5]: Good use of mocking for testing loading data into our RDS.

File: weekly-openmeteo/conftest.py [hunk 2]
```diff
@@ -108,3 +113,10 @@ def invalid_openmeteo_data():
             'sunset': ['2024-10-09T18:00', '2024-10-10T18:01', '2024-10-11T18:02',
                        '2024-10-12T18:03', '2024-10-13T18:04', '2024-10-14T18:05',
                        '2024-10-15T18:06']}}]*100
+
+
+@pytest.fixture
+def valid_insertion_data():
+    '''Returns a valid dataset for insertion'''
+    return [[1, '2024-10-09 06:00:00', '2024-10-09 18:00:00'],
+            [2, '2024-10-10 06:01:00', '2024-10-10 18:01:00']]
```
Comment (MurasatoNaoya) [inline weekly-openmeteo/conftest.py:122]: An strong, almost exhaustive array of fixtures to be used for mocking test functions.

File: weekly-openmeteo/load.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+from dotenv import load_dotenv
+from extract import get_connection
+
+def load_data(data: list[list]) -> None:
+    '''Inserts data into a postgres database'''
+    query = '''INSERT INTO solar_feature (county_id,sunrise_timestamp,sunset_timestamp)
+    VALUES (%s,%s,%s)'''
+    with get_connection() as conn:
+        with conn.cursor() as cur:
+            cur.executemany(query, data)
```
Comment (MurasatoNaoya) [inline weekly-openmeteo/load.py:12]: Good use of bulk insertion of flattened data. This efficiency will help us stay with the 15 minute ceiling of our lambda functions.

**Review by MurasatoNaoya — APPROVED** [review]
LGTM.

**Pull request merged**